agent/agent.go

```diff
@@ -1217,6 +1196,33 @@ func (a *Agent) listenHTTP() ([]*HTTPServer, error) {
 	return servers, nil
 }
 
+// setupHTTPServer adds HTTP/2 support, ConnState, and a connection handshake timeout
```

Comment:
What if this was named `setupHTTPS`? Since it seems to be only for HTTPS.

Reply:
Good suggestion, I like that name better.